pkg/remote/client.go

```diff
@@ -0,0 +1,84 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	prompb "go.buf.build/grpc/go/prometheus/prometheus"
+)
+
+type HTTPConfig struct {
+	UserAgent string
+	Timeout   time.Duration
+	TLSConfig tls.Config
+	BasicAuth BasicAuth
+	Headers   http.Header
+}
+
+type BasicAuth struct {
+	User, Password string
+}
+
+type WriteClient struct {
+	hc  *http.Client
+	url *url.URL
+	cfg *HTTPConfig
```

Comment:
:thinking: Maybe I'm missing something, but where do we actually use it? I mean, let's say the endpoint requires basic auth or TLS config, or even we'd like to pass the headers.

Reply:
Good catch, I forgot to include this.I will do.